docs/markdown/Writing Plugins/tutorials/01-create-new-goal-with-a-custom-target.md

```diff
@@ -0,0 +1,364 @@
+]
+```
+
+## Creating a new target
+
+Once you have become familiar with the [core concepts of Targets and Fields](https://www.pantsbuild.org/docs/target-api-concepts), you are ready to [create an own custom target](https://www.pantsbuild.org/docs/target-api-new-targets) that will represent the `VERSION` file:
+
+```python pants-plugins/project_version/targets.py
+from pants.engine.target import COMMON_TARGET_FIELDS, SingleSourceField, Target
+
+
+class ProjectVersionTarget(Target):
+    alias = "version_file"
+    core_fields = (*COMMON_TARGET_FIELDS, SingleSourceField)
+    help = "A project version target representing the VERSION file."
+```
+
+Our target has some common target fields such as `tags` and `description` available via the `COMMON_TARGET_FIELDS`; including those fields in your targets may be convenient if you decide to use tags and provide a description later. In addition, it also has the [`source` field](https://www.pantsbuild.org/docs/rules-api-and-target-api#sourcesfield) which will be used to provide path to the project's `VERSION` file. 
+
+We could [add a custom field](https://www.pantsbuild.org/docs/target-api-new-fields) to provide a file path, however, there are multiple advantages to  using the `source` field. You will learn more about them in the following tutorials. 
+
+In order to start using a target, you only need to register it: 
+
+```python pants-plugins/project_version/register.py
+from typing import Iterable
+
+from pants.engine.target import Target
+from project_version.target_types import ProjectVersionTarget
+
+
+def target_types() -> Iterable[type[Target]]:
+    return [ProjectVersionTarget]
+```
+
+You can now run `./pants help version_file` to learn more about the target. You can now also add a target to a `myapp/BUILD` file:
+
+```python
+version_file(
+    name="main-project-version", 
+    source="VERSION",
+)
+```
+
+Since you have registered the target, Pants will be able to "understand" it:
+
+```text
+$ ./pants peek myapp:main-project-version
+[
+  {
+    "address": "myapp:main-project-version",
+    "target_type": "version_file",
+    "dependencies": [],
+    "description": null,
+    "source_raw": "VERSION",
+    "sources": [
+      "myapp/VERSION"
+    ],
+    "tags": null
+  }
+]
+```
+
+## Creating a goal
+
+[Goals](https://www.pantsbuild.org/docs/goals) are the commands that Pants runs such as `fmt` or `lint`. To be able to get a project version number (using a target we've just created), we need to [create a new goal](https://www.pantsbuild.org/docs/rules-api-goal-rules). The code below is the boilerplate necessary to create a goal, so it's not really necessary to understand how, for instance, subsystems work right now. The function decorated with the `@goal_rule` can be named anything, but it's helpful for the name to represent the functionality your goal provides. To make your goal part of the plugin's interface, add it to the `rules` function in the `register.py` module.
```

Comment:
Is it worth a note here about whether a typical user would need to create a goal rule plugin (v.s. a plugin for other goals?). 

Maybe a one or two-liner?

I've made about 20 plugins - but not a goal plugin yet 🤷🏽

Reply:
Excellent idea, done!